### PR TITLE
Fix SwitchedLikelihood vs num_latent_gps bug

### PR DESCRIPTION
**Pull request by st--**
Rescues PR #957 which fixed bug #951.

### CODE REVIEW

**Comment by awav** [comment]
Could you please link the issue from our project board to that issue? Thanks

File: gpflow/models/model.py [hunk 1]
```diff
@@ -103,6 +104,32 @@ def __init__(
...
+        _, Y = data
+        output_dim = Y.shape[1]
+        return GPModel.num_latent_gps_for_likelihood_and_output_dim(likelihood, output_dim)
+
+    @staticmethod
+    def num_latent_gps_for_likelihood_and_output_dim(likelihood, output_dim) -> int:
+        """
+        Note: It's not nice for `GPModel` to need to be aware of specific
+        likelihoods as here. However, `num_latent_gps` is a bit more broken in
+        general, specifically regarding multioutput kernels. We should fix this
```
Comment (st--) [inline gpflow/models/model.py]: check that we've got a separate issue to keep track of this; add issue url to code

Comment (st--) [reply gpflow/models/model.py]: @condnsdmatters I merged your suggestion in here. Do you think we can remove this point from the docstring now ?

File: gpflow/models/model.py [hunk 2]
```diff
@@ -103,6 +104,36 @@ def __init__(
...
         self.likelihood = likelihood
 
+    @staticmethod
+    def calc_num_latent_gps_from_data(data, kernel: Kernel, likelihood: Likelihood) -> int:
+        _, Y = data
+        output_dim = Y.shape[-1]
+        return GPModel.calc_num_latent_gps(kernel, likelihood, output_dim)
+
+    @staticmethod
+    def calc_num_latent_gps(kernel: likelihood: Likelihood, Kernel, output_dim: int) -> int:
```
Comment (condnsdmatters) [inline gpflow/models/model.py]: looks like some vim tricks went wrong ;)
```suggestion
    def calc_num_latent_gps(kernel: Kernel, likelihood: Likelihood, output_dim: int) -> int:
```

**Pull request merged**